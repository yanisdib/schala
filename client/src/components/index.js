// COMPONENTS EXPORTS

export { default as Header } from './Header/Header';

export { default as Button } from './Button/Button';
export { default as MenuButton } from './MenuButton/MenuButton';
export { default as DisplayIdentifiedButton } from './DisplayIdentifiedButton/DisplayIdentifiedButton';

export { default as Tag } from './Tag/Tag';
export { default as Pointer } from './Pointer/Pointer';
export { default as Bubble } from './Bubble/Bubble';

export { default as PictureIdentifier } from './PictureIdentifier/PictureIdentifier';