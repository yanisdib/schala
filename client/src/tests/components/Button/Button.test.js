import { shallow } from 'enzyme';
import { Button } from '../../../components';

describe('Button Component', () => {
    let wrapper;
    let defaultProps = Button.defaultProps.variant;

    beforeEach(() => wrapper = shallow(<Button variant='primary'>Button</Button>));

    it('renders properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders with default props properly', () => {
        expect(defaultProps).toEqual('primary');
    });

    it('renders with children text properly', () => {
        expect(wrapper.text()).toEqual('Button');
    });
});