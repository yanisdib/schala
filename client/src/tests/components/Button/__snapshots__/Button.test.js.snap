// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders properly 1`] = `
<button
  className="sc-bdfBQB jJtPLv"
>
  Button
</button>
`;
